### PR TITLE
Add book to the user borrowing list

### PR DESCRIPTION
**Pull request by lubnaabd**
## Description

Add available books from library to the user borrowing list  front-end and back-end code

### Type of change

New features #79 
Description : 
* add style for the borrow section
* create add_book_to_user that contain fetch function that make a request (post) to back-end and contain validation function.

New features #80 
Description : 
* create addBookToUser  in the controller add  book to user.

New features #18
Description :
- create get_library_id.js query file.
- create set_borrow_book.js query file.
- create check_library_id.js query file

New features testing

- test route add book o user.
- test get_library_id.js  query.
- test set_borrow_book.js query 
- test check_library_id.js query

### CODE REVIEW

**Review by ali-7 — APPROVED** [review]

File: public/js/add_book_to_user.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    body: JSON.stringify(data),
+  })
+    .then(result => result.json())
+    .then((response) => {
+      if (response.message === ' هذا الكتاب مستعار ') {
+        swal('خطأ !', response.message, 'error').then((value) => {
+          JSON.stringify({ response });
+        });
+      } else {
+        console.log(response);
```
Comment (RamyAlshurafa) [inline public/js/add_book_to_user.js]: `console.log` !!

File: public/js/add_book_to_user.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    method: 'POST',
+    credentials: 'same-origin',
+    headers: { 'Content-Type': 'application/json; charset=utf-8' },
+    body: JSON.stringify(data),
+  })
+    .then(result => result.json())
+    .then((response) => {
+      if (response.message === ' هذا الكتاب مستعار ') {
+        swal('خطأ !', response.message, 'error').then((value) => {
+          JSON.stringify({ response });
```
Comment (RamyAlshurafa) [inline public/js/add_book_to_user.js]: `JSON.stringify({ response })` why ?

File: public/js/add_book_to_user.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        console.log(response);
+        if (response.resultLibrary.length > 0) {
+          const href = window.location.href;
+          const mobileNum = href.split('=')[1];
+          swal('إضافة', response.message, 'success').then((value) => {
+            window.location = `/admin/borrow?data=${mobileNum}`;
+          });
+        } else {
+          swal('خطأ !', response.message, 'error').then((value) => {
+            JSON.stringify({ response });
```
Comment (RamyAlshurafa) [inline public/js/add_book_to_user.js]: same here, what is the purpose of this line?

File: src/controllers/borrow.js [hunk 4]
```diff
@@ -47,21 +50,45 @@ exports.addUser = (req, response, next) => {
   const userData = req.body;
   setUser(userData)
     .then((results) => {
-      const result = { message: 'User Added !' };
+      const mobileNumberUser = results[0].mobileNumber;
+      const result = { message: 'User Added !', mobileNumberUser };
       response.send(JSON.stringify(result));
```
Comment (RamyAlshurafa) [inline src/controllers/borrow.js]: you can use `res.json` instead
you can read about it and use it if you want

File: src/controllers/borrow.js [hunk 5]
```diff
@@ -47,21 +50,45 @@ exports.addUser = (req, response, next) => {
...
     })
     .catch((err) => {
       next(err);
```
Comment (RamyAlshurafa) [inline src/controllers/borrow.js]: `next(err)` will render server error page(`HTML`) while your request sent by  `fetch` so you cant render `HTML` files

**Review by RamyAlshurafa — CHANGES_REQUESTED** [review]

**Review by RamyAlshurafa — APPROVED** [review]
#### Good Job
but I think you should split this PR into front-end & back-end and make 2 PR

**Pull request merged**